controllers/machinedeletionremediation_controller.go

```diff
@@ -90,62 +90,46 @@ func (r *MachineDeletionRemediationReconciler) Reconcile(ctx context.Context, re
 		return ctrl.Result{}, err
 	}
 
-	annotations := remediation.GetAnnotations()
-	if annotations != nil {
-		machineNameNamespace, exists := annotations[MachineNameNamespaceAnnotation]
-		if exists {
-			machineName, machineNamespace, err := extractNameAndNamespace(machineNameNamespace, remediation.GetName())
-			if err != nil {
-				log.Error(err, "could not get Machine data from remediation", "remediation", remediation.GetName(), "annotation", machineNameNamespace)
-				return ctrl.Result{}, nil
-			}
+	log.Info("Machine Deletion Remediation CR found", "name", remediation.GetName())
 
-			ok, err := r.verifyMachineIsDeleted(ctx, machineName, machineNamespace, remediation.Name)
-			if err != nil {
-				return ctrl.Result{}, err
-			}
-			if !ok {
-				return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
-			}
+	var machine *unstructured.Unstructured
+	if machine, err = r.getMachine(remediation); err != nil {
+		if apiErrors.IsNotFound(err) {
 			return ctrl.Result{}, nil
 		}
-	}
-
-	var node *v1.Node
-	if node, err = r.getNodeFromMdr(remediation); err != nil {
+		log.Error(err, "Unexpected error fetching Machine from Node", "node", remediation.Name)
 		return ctrl.Result{}, err
 	}
 
-	log.Info("MDR CR and affected Node found", "node", node.GetName())
+	log.Info("node-associated machine found", "node", remediation.Name, "machine", machine.GetName())
 
-	var machine *unstructured.Unstructured
-	if machine, err = r.buildMachineFromNode(node); err != nil {
-		r.Log.Error(err, noMachineFoundError, "node name", node.Name)
-		return ctrl.Result{}, err
-	}
-
-	if !hasControllerOwner(machine) {
-		log.Info("ignoring remediation of node-associated machine: the machine has no controller owner", "machine", machine.GetName(), "node name", remediation.Name)
-		return ctrl.Result{}, nil
-	}
-
-	log.Info("node-associated machine found", "node", node.Name, "machine", machine.GetName())
+	if !isMachineMarkedForDeletion(machine) {
+		if !hasControllerOwner(machine) {
+			log.Info("ignoring remediation of node-associated machine: the machine has no controller owner", "machine", machine.GetName(), "node name", remediation.Name)
+			return ctrl.Result{}, nil
+		}
 
-	err = r.deleteMachineOfNode(ctx, machine, remediation.Name)
-	if err != nil {
-		return ctrl.Result{}, err
-	}
+		// save Machine's name and namespace to follow its deletion phase when its related node is deleted
+		if err = r.saveMachineNameNs(ctx, remediation, machine.GetName(), machine.GetNamespace()); err != nil {
```

Comment:
Nit: since we are already using the MarkedForDeletion terminology, I think that  `markForDeletion` would be more suitable than `saveMachineNameNs`

Reply:
disagree: the name explains what the function does correctly. What we make out of it at a later stage is another story IMHO

Reply:
Yep, I agree - it was my mistake of understanding the code - initially I've wrongly assumed that we use the annotation set on the remediation as a "Marker" which will later will be used indicate whether to delete the machine or not.